auraed/src/cells/cell_service/cell_service.rs

```diff
@@ -237,6 +241,29 @@ impl CellService {
         executables.broadcast_stop().await;
         Ok(())
     }
+
+    #[tracing::instrument(skip(self))]
+    async fn list(&self) -> Result<CellServiceListResponse> {
+        let cells = self.cells.lock().await;
+
+        
+        Ok(CellServiceListResponse { 
+            cells: cells.entries()
+                        .iter()
+                        .map(|cell| 
+                            Cell { 
+                                name: cell.name().to_string(), 
+                                cpu: Some(CpuController { 
+                                    weight: cell.spec().cgroup_spec.cpu.as_ref().and_then(|cpu| cpu.weight.as_ref().map(|w| w.into_inner())), 
```

Comment:
Instead of having `cell.spec().cgroup_spec.cpu.as_ref()` multiple times, I suggest making a variable to hold the reference.

Reply:
i think i'd pull the map closure out to a function so it can be tested in isolation too.

Reply:
I've extracted it for now. Will probably extract this into separate functions when I have the nested cells in there as well.